packages/flutter/lib/src/widgets/slotted_render_object_widget.dart

```diff
@@ -211,34 +220,67 @@ class SlottedRenderObjectElement<S> extends RenderObjectElement {
   @override
   void mount(Element? parent, Object? newSlot) {
     super.mount(parent, newSlot);
-    _updateChildren();
+    _updateChildren(widget as SlottedMultiChildRenderObjectWidgetMixin<S>);
   }
 
   @override
   void update(SlottedMultiChildRenderObjectWidgetMixin<S> newWidget) {
     super.update(newWidget);
     assert(widget == newWidget);
-    _updateChildren();
+    _updateChildren(newWidget);
   }
 
   List<S>? _debugPreviousSlots;
 
-  void _updateChildren() {
+  void _updateChildren(SlottedMultiChildRenderObjectWidgetMixin<S> newWidget) {
     final SlottedMultiChildRenderObjectWidgetMixin<S> slottedMultiChildRenderObjectWidgetMixin = widget as SlottedMultiChildRenderObjectWidgetMixin<S>;
     assert(() {
       _debugPreviousSlots ??= slottedMultiChildRenderObjectWidgetMixin.slots.toList();
       return listEquals(_debugPreviousSlots, slottedMultiChildRenderObjectWidgetMixin.slots.toList());
     }(), '${widget.runtimeType}.slots must not change.');
     assert(slottedMultiChildRenderObjectWidgetMixin.slots.toSet().length == slottedMultiChildRenderObjectWidgetMixin.slots.length, 'slots must be unique');
 
+    final Map<GlobalKey, S> globalKeyedSlots = _globalKeyToSlot;
+    _globalKeyToSlot = <GlobalKey, S>{};
+
+    // New slots for children that are moved to a different slot by global key
+    // reparenting. It shouldn't be updated in place because two children could
+    // swap slots.
+    final Map<S, Element> slotToKeyedChild = <S, Element>{};
+
+    // Process children that need to be moved to new slots because of global
+    // key reparenting first.
     for (final S slot in slottedMultiChildRenderObjectWidgetMixin.slots) {
-      _updateChild(slottedMultiChildRenderObjectWidgetMixin.childForSlot(slot), slot);
+      final Widget? widget = slottedMultiChildRenderObjectWidgetMixin.childForSlot(slot);
+      final Key? newWidgetKey = widget?.key;
+      if (widget != null && newWidgetKey is GlobalKey) {
```

Comment:
This now has me wondering: Don't we need to do this for all keys, not just globalkeys? If a keyed widget changes slots, it should be matched with the element from the previous slot it was in?